src/main/java/net/apnic/rdap/filter/filters/RDAPPreFilter.java

```diff
@@ -40,6 +42,8 @@ public Object run()
         RDAPRequestPath path = RDAPRequestPath.createRequestPath(
                 context.getRequest().getRequestURI());
         context.put(RequestContextKeys.RDAP_REQUEST_PATH, path);
+        context.getRequest().setAttribute(ORIGIN_ATT_KEY,
+            context.getRequest().getRequestURL().toString());
```

Comment:
Most likely this may disappear with your other comments above. Will advice what happens